File: DistributedComputingAndSystems/Project/code-python/code/codes.py
from enum import Enum, auto

class CODES(Enum):
    # example RED = "red"
    HELLO = "hello"
    HELLO_RPL = "hello_rpl"
    TEXT = "text"
    EXIT = "exit"
    INVITE = "invite"
    ACC_INVITE = "acc_invite"
    START = "start"
    LEADER = "leader"
    ON_MOVE = "on_move"
    PLAYED_MOVE = "played move"
    WINNER = "winner"
    RETURN = "return"
    UPDATE_STATE = "update_state"
    TEST = "test"
    

    def encode(stringName):
        for code in CODES:
            if code.value == stringName:
                return code
        return None

    def decode(self):
        return self.value
    
    def __str__(self) -> str:
        return str(self.value)

File: DistributedComputingAndSystems/Project/DSC_Project_java/src/wut/weiti/edcs/ludo/Codes.java
package wut.weiti.edcs.ludo;

public enum Codes {
    HELLO
    {
        @Override
        public String toString() {
            return "hello";
        }
    },
    HELLO_RPL{
        @Override
        public String toString() {
            return "hello_rpl";
        }
    },
    TEXT{
        @Override
        public String toString() {
            return "text";
        }
    },
    EXIT{
        @Override
        public String toString() {
            return "exit";
        }
    },
    INVITE{
        @Override
        public String toString() {
            return "invite";
        }
    },
    ACC_INVITE{
        @Override
        public String toString() {
            return "acc_invite";
        }
    },
    START{
        @Override
        public String toString() {
            return "start";
        }
    },
    LEADER{
        @Override
        public String toString() {
            return "leader";
        }
    },
    ON_MOVE{
        @Override
        public String toString() {
            return "on_move";
        }
    },
    PLAYED_MOVE{
        @Override
        public String toString() {
            return "played_move";
        }
    },
    WINNER{
        @Override
        public String toString() {
            return "winner";
        }
    },
    RETURN{
        @Override
        public String toString() {
            return "return";
        }
    },
    UPDATE_STATE{
        @Override
        public String toString() {
            return "update_state";
        }
    },
    TEST{
        @Override
        public String toString() {
            return "test";
        }
    },
}
